presto-main/src/main/java/io/prestosql/connector/system/TableCommentSystemTable.java

```diff
@@ -0,0 +1,124 @@
+
+    private static final ConnectorTableMetadata COMMENT_TABLE = tableMetadataBuilder(COMMENT_TABLE_NAME)
+            .column("catalog_name", createUnboundedVarcharType())
+            .column("schema_name", createUnboundedVarcharType())
+            .column("table_name", createUnboundedVarcharType())
+            .column("comment", createUnboundedVarcharType())
+            .build();
+
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public TableCommentSystemTable(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return COMMENT_TABLE;
+    }
+
+    @Override
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession connectorSession, TupleDomain<Integer> constraint)
+    {
+        Optional<String> catalogFilter = stringFilter(constraint, 0);
+        Optional<String> schemaFilter = stringFilter(constraint, 1);
+        Optional<String> tableFilter = stringFilter(constraint, 2);
+
+        Session session = toSession(transactionHandle, connectorSession);
+        Builder table = InMemoryRecordSet.builder(COMMENT_TABLE);
+
+        for (String catalog : filter(listCatalogs(session, metadata, accessControl).keySet(), catalogFilter)) {
+            QualifiedTablePrefix prefix = tablePrefix(catalog, schemaFilter, tableFilter);
+
+            Set<SchemaTableName> names = ImmutableSet.of();
+            try {
+                names = listTables(session, metadata, accessControl, prefix);
+            }
+            catch (PrestoException ignored) {
+                // listTables throws an exception if cannot connect the database
+            }
+
+            for (SchemaTableName name : names) {
+                QualifiedObjectName tableName = new QualifiedObjectName(prefix.getCatalogName(), name.getSchemaName(), name.getTableName());
+                Optional<String> comment = Optional.empty();
+                try {
+                    comment = metadata.getTableHandle(session, tableName)
+                                      .map(handle -> metadata.getTableMetadata(session, handle))
+                                      .map(metadata -> metadata.getMetadata().getComment())
+                                      .get();
+                }
+                catch (PrestoException | NullPointerException ignored) {
+                    // getTableHandle may throw an exception
+                    // Cassandra connector throws PrestoException if the table has case insensitive column names
+                    // Cassandra connector throws NullPointerException if the table has unsupported column types
```

Comment:
Can we somehow fix this, logic that relies on NPE sounds...

Reply:
I'll fix it in this PR https://github.com/prestosql/presto/pull/592.